topdown/eval.go

```diff
@@ -1682,30 +1685,62 @@ func (e evalBuiltin) eval(iter unifyIterator) error {
 	numDeclArgs := len(e.bi.Decl.FuncArgs().Args)
 
 	e.e.instr.startTimer(evalOpBuiltinCall)
+	var err error
+	needNormalUnify := true
 
-	err := e.f(e.bctx, operands, func(output *ast.Term) error {
-
+	// We skip evaluation of the builtin entirely if the NDBuiltinCache is
```

Comment:
Does this mean we can clean up the other places where we've done that using the bctx's Cache? E.g. [rand.intn here](https://github.com/open-policy-agent/opa/blob/39fa5cf79caec81433dba1770346b57c932a6414/topdown/numbers.go#L80-L84)

Reply:
I'm not 100% sure, but I think that logic may still need to stick around. :thinking:

The ND builtin cache is opt-in, the original `bctx` cache is mandatory. The two are separate systems for now, but if the ND builtin cache was mandatory too, I could see doing some replacing of logic...

Reply:
💡 Ah, right. I must have forgotten about the optional part. That's fair.